clients/src/main/java/org/apache/kafka/clients/telemetry/DefaultClientTelemetry.java

```diff
@@ -359,6 +360,18 @@ public void telemetrySubscriptionSucceeded(GetTelemetrySubscriptionsResponseData
         }
     }
 
+    private int getPushIntervalMs(final GetTelemetrySubscriptionsResponseData data) {
+        int pushIntervalMs = validatePushIntervalMs(data.pushIntervalMs() > 0 ? data.pushIntervalMs() : 10000);
+        if (!subscription().isPresent()) {
+            // if this is the first request, subscription() returns null, and we want to
+            // equally spread all the request between 0.5 pushInterval and 1.5 pushInterval.
+            final double rand = ThreadLocalRandom.current().doubles(1, 0, 1).findFirst().orElse(1.0);
```

Comment:
I _think_ this might be doable in a more straightforward way:

```suggestion
            double rand = ThreadLocalRandom.current().nextDouble(0.5, 1.5);
            return (int) Math.round(rand * pushIntervalMs);
```

Reply:
way better this way.